internal/api/assignments.go

```diff
@@ -0,0 +1,107 @@
+package api
+
+import (
+	"net/http"
+
+	"go.infratographer.com/permissions-api/internal/types"
+
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+	"go.infratographer.com/x/urnx"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+func (r *Router) assignmentCreate(c *gin.Context) {
+	roleIDStr := c.Param("role_id")
+
+	roleID, err := uuid.Parse(roleIDStr)
+	if err != nil {
+		c.JSON(http.StatusNotFound, gin.H{"message": "not found"})
+		return
+	}
+
+	ctx, span := tracer.Start(c.Request.Context(), "api.assignmentCreate", trace.WithAttributes(attribute.String("role_id", roleIDStr)))
+	defer span.End()
+
+	var reqBody createAssignmentRequest
+
+	err = c.BindJSON(&reqBody)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"message": "error parsing request body", "error": err.Error()})
+		return
+	}
+
+	subjURN, err := urnx.Parse(reqBody.SubjectURN)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"message": "error parsing subject URN", "error": err.Error()})
+		return
+	}
+
+	subjResource, err := r.engine.NewResourceFromURN(subjURN)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"message": "error creating resource", "error": err.Error()})
+		return
+	}
+
+	role := types.Role{
+		ID: roleID,
+	}
+
+	_, err = r.engine.AssignSubjectRole(ctx, subjResource, role)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"message": "error creating resource", "error": err.Error()})
+		return
+	}
+
+	resp := createAssignmentResponse{
+		Success: true,
```

Comment:
There's no assignment ID as such. Assignments are relationships, and the entire (resource, relation, subject) tuple effectively is the identifier for a relationship, where SpiceDB only really supports querying for sets of relationships by certain criteria.

In other words, rather than saying "delete assignment `$UUID`", you say "delete all relationships with a resource of _x_, a relation of _y_, and (maybe) a subject resource of _z_".